StreamCryptor/Helper/Utils.cs

```diff
@@ -102,36 +102,67 @@ public static string GetChecksum(byte[] array)
         ///     Generates random number.
         /// </summary>
         /// <param name="maxNumber">The max number.</param>
-        /// <see cref="http://blog.codeeffects.com/Article/Generate-Random-Numbers-And-Strings-C-Sharp" />
         /// <returns>A random number.</returns>
         /// <exception cref="ArgumentOutOfRangeException"></exception>
+        /// <remarks>Move to libsodium-net: waiting for stable release</remarks>
         public static int GetRandomNumber(int maxNumber)
```

Comment:
Maybe something like this would be faster?

```csharp
  public static uint GetRandomNumber(uint maxNumber)
  {

    if (maxNumber < 2) {
        return 0;
    }
    var min = (~maxNumber + 1) % maxNumber;
    var b = new byte[sizeof(uint)];
    var rng = new RNGCryptoServiceProvider();
    uint r;
    do {
        rng.GetBytes(b);
        r = BitConverter.ToUInt32(b, 0);
    } while (r < min);
    return r % maxNumber;
  }
```